### ISSUE TITLE
Android Studio: flutter.gradle fails when attempting to run in release mode on connected device (no option named "track-widget-creation")

### ISSUE THREAD

**Issue by sir-boformer** [flutter/flutter#20260]
Happens with all of my Flutter projects. Running in debug mode works finde, and `flutter build apk --release` also runs through without a problem.

Flutter version: Latest master

How to reproduce:

1. Create a new Flutter project in Android Studio
2. Attach Android device
3. Add `--release` flag as additional argument in the run configuration
4. Run on device (green play button)

Console output in Android Studio:

```
Initializing gradle...
Resolving dependencies...
Launching lib\main.dart on BAH L09 in release mode...
Running 'gradlew assembleRelease'...
Could not find an option named "track-widget-creation".

Run 'flutter -h' (or 'flutter <command> -h') for available flutter commands and options.

FAILURE: Build failed with an exception.

* Where:
Script 'C:\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 413

* What went wrong:
Execution failed for task ':app:flutterBuildRelease'.
> Process 'command 'C:\flutter\bin\flutter.bat'' finished with non-zero exit value 64

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s
Finished with error: Gradle build failed: 1

```

**Comment by sir-boformer** [comment]
To make it work, these 3 lines must be removed from `flutter.gradle`:

https://github.com/flutter/flutter/blob/8c02b8f889d1012c4e1c74e8e8df919f64d9e0a0/packages/flutter_tools/gradle/flutter.gradle#L431-L433

**Comment by sir-boformer** [comment]
Fixed by Flutter Intellij plugin 27.1 (change the preference for --track-widget-creation to default to off)

**Comment by rgisi** [comment]
Hi, this just happened again to me. 
Flutter Plugin Version is 30.0.1 and the flutter.gradle file contains these lines:
```
Boolean trackWidgetCreationValue = false
if (project.hasProperty('track-widget-creation')) {
   trackWidgetCreationValue = project.property('track-widget-creation').toBoolean()
}
```
The project itself doesn't mention the parameter "track-widget-creation" anywhere.
Doesn't happen when I build with the flutter console.

**Comment by zoechi** [comment]
@rgisi Can you reproduce in a newly create (with recent Flutter version) project?
You could try deleting `ios/` and `android/` and run `flutter create .` in your project directory (and reapply manual changes)

**Comment by rgisi** [comment]
Yes. I just created a new project without any modifications (asides from code in main.dart), same behaviour: Debug runs smoothly, Release cannot be build with:
```
Initializing gradle...
Resolving dependencies...
Launching lib\main.dart on LG H870 in release mode...
Gradle task 'assembleRelease'...
Could not find an option named "track-widget-creation".

Run 'flutter -h' (or 'flutter <command> -h') for available flutter commands and options.
Finished with error: Gradle task assembleRelease failed with exit code 1
```

**Comment by zoechi** [comment]
Merging with #24239 which is open.

**Comment by zoechi** [comment]
@rgisi you could try if `flutter run --no-track-widget-creation` is supported.

**Comment by rgisi** [comment]
That works, thanks!

**Comment by Gyorgio88** [comment]
I guys,
i had the same problem on android studio.
with `--release and --no-track-widget-creation flags`, when run on device, i have this result:
 
Initializing gradle...
Resolving dependencies...
Launching lib\main.dart on LG G2 in release mode...
Gradle task 'assembleRelease'...

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForRelease'.
> java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 1 column 1 path $

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

**Comment by zoechi** [comment]
@Gyorgio88 this issue is closed, please follow #24239

**Comment by Gyorgio88** [comment]
> 
> 
> @Gyorgio88 this issue is closed, please follow #24239

pardon.
thanks
